sycl/source/detail/usm/usm_impl.cpp

```diff
@@ -70,8 +71,32 @@ void *alignedAllocHost(size_t Alignment, size_t Size, const context &Ctxt,
 
     switch (Kind) {
     case alloc::host: {
-      Error = Plugin.call_nocheck<PiApiKind::piextUSMHostAlloc>(
-          &RetVal, C, nullptr, Size, Alignment);
+      const platform &Platform = Ctxt.get_platform();
+
+      std::vector<pi_usm_mem_properties> Props;
+
+      if (Platform.has_extension("cl_intel_mem_alloc_buffer_location") &&
+          PropList.has_property<cl::sycl::ext::intel::experimental::property::
+                                    usm::buffer_location>()) {
+        Props.emplace_back(PI_MEM_USM_ALLOC_BUFFER_LOCATION);
+        Props.emplace_back(
+            PropList
+                .get_property<cl::sycl::ext::intel::experimental::property::
+                                  usm::buffer_location>()
+                .get_buffer_location());
+      }
+
+      if (Props.empty()) {
+        // Explicitly pass null pointer since the OpenCL driver may not support
+        // empty property lists, i.e., containing only a null terminator.
+        Error = Plugin.call_nocheck<PiApiKind::piextUSMHostAlloc>(
+            &RetVal, C, nullptr, Size, Alignment);
+      } else {
+        Props.emplace_back(0); // null-terminate property list
+        Error = Plugin.call_nocheck<PiApiKind::piextUSMHostAlloc>(
+            &RetVal, C, Props.data(), Size, Alignment);
+      }
+
```

Comment:
the spec https://github.com/sherry-yuan/llvm/blob/9301c51e755d118cee94fb2df391ccb3549d6a23/sycl/doc/extensions/proposed/sycl_ext_intel_runtime_buffer_location.asciidoc says:

> this property has no effect when passed to sycl::malloc_shared() or sycl::malloc_host()

so why is this code not ignoring it?

Reply:
Spec is being updated to not say that: https://github.com/intel/llvm/pull/6219

Reply:
The spec is being updated to not say that: https://github.com/intel/llvm/pull/6219